File: core/src/main/java/ru/surfstudio/android/core/ui/base/navigation/dialog/route/DialogRoute.java
package ru.surfstudio.android.core.ui.base.navigation.dialog.route;


import android.support.v4.app.DialogFragment;

import ru.surfstudio.android.core.ui.base.navigation.fragment.route.FragmentRoute;

/**
 * см {@link FragmentRoute}
 */
public abstract class DialogRoute extends FragmentRoute {

    @Override
    protected abstract Class<? extends DialogFragment> getFragmentClass();

    @Override
    public DialogFragment createFragment(){
        return (DialogFragment)super.createFragment();
    }
}

File: navigation/lib-navigation/src/main/java/ru/surfstudio/android/navigation/route/dialog/DialogRoute.kt
package ru.surfstudio.android.navigation.route.dialog

import androidx.fragment.app.DialogFragment
import ru.surfstudio.android.navigation.route.BaseRoute

/**
 * Route for [DialogFragment].
 */
open class DialogRoute : BaseRoute<DialogFragment>() {

    /**
     * Creates DialogFragment with [getScreenClass] or [getScreenClassPath] class.
     */
    open fun createDialog(): DialogFragment {
        return requireScreenClass().newInstance()
                .apply { arguments = prepareDataWithId() }
    }
}